clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -267,11 +267,13 @@ public static void waitForCondition(final TestCondition testCondition, final Str
     public static void waitForCondition(final TestCondition testCondition, final long maxWaitMs, String conditionDetails) throws InterruptedException {
         final long startTime = System.currentTimeMillis();
 
-        while (!testCondition.conditionMet() && ((System.currentTimeMillis() - startTime) < maxWaitMs)) {
+        boolean testConditionMet = testCondition.conditionMet();
+        while (!testConditionMet && ((System.currentTimeMillis() - startTime) < maxWaitMs)) {
```

Comment:
This can be written as:

```java
        boolean conditionMet = false;
        while (!(conditionMet = testCondition.conditionMet()) && ((System.currentTimeMillis() - startTime) < maxWaitMs))
            Thread.sleep(Math.min(maxWaitMs, 100L));
```